src/pages/Defi/views/StakingVaults.tsx

```diff
@@ -39,12 +44,82 @@ const DefiHeader = () => {
 }
 
 const FoxFarmCTA = () => {
+  const dispatch = useAppDispatch()
   const translate = useTranslate()
   const history = useHistory()
   const location = useLocation()
+
+  const [lpApy, setLpApy] = useState<Nullable<string>>(null)
+  const [farmingV4Data, setFarmingV4Data] =
+    useState<Nullable<GetFoxFarmingContractMetricsReturn>>(null)
+  const [isLpAprLoaded, setIsLpAprLoaded] = useState<boolean>(false)
+  const [isFarmingAprV4Loaded, setIsFarmingAprV4Loaded] = useState<boolean>(false)
+
+  const ethAccountIds = useAppSelector(state =>
+    selectAccountIdsByAssetId(state, { assetId: ethAssetId }),
+  )
+
+  useEffect(() => {
+    ;(async () => {
+      if (!ethAccountIds?.length) return
+
+      const ethAccountAddresses = ethAccountIds.map(accountId => fromAccountId(accountId).account)
+
+      const metricsPromises = await Promise.all(
+        ethAccountAddresses.map(
+          async accountAddress =>
+            await dispatch(
+              foxEthApi.endpoints.getFoxEthLpMetrics.initiate({
+                accountAddress,
+              }),
+            ),
+        ),
+      )
+
+      // To get the APY, we need to fire the metrics requests for all accounts
+      // However, it doesn't matter which account we introspect - it's going to be the same for all accounts
+      const { isLoading, isSuccess, data } = metricsPromises[0]
+
+      if (isLoading || !data) return
+
+      if (isSuccess) {
+        setLpApy(data.apy)
+        setIsLpAprLoaded(true)
+      }
+    })()
+  }, [ethAccountIds, dispatch])
+
+  useEffect(() => {
+    ;(async () => {
+      if (!ethAccountIds?.length) return
+
+      const ethAccountAddresses = ethAccountIds.map(accountId => fromAccountId(accountId).account)
+
+      const metricsPromises = await Promise.all(
+        ethAccountAddresses.map(
+          async accountAddress =>
+            await dispatch(
+              foxEthApi.endpoints.getFoxFarmingContractMetrics.initiate({
+                contractAddress: FOX_FARMING_V4_CONTRACT_ADDRESS,
+                accountAddress,
+              }),
+            ),
+        ),
+      )
+
+      // To get the FOX farming contract metrics data, it doesn't matter which account we introspect - it's going to be the same for all accounts
+      const { isLoading, isSuccess, data } = metricsPromises[0]
+
+      if (isLoading || !data) return
+
+      if (isSuccess) {
+        setFarmingV4Data(data)
+        setIsFarmingAprV4Loaded(true)
+      }
+    })()
+  }, [ethAccountIds, dispatch])
```

Comment:
This who section is duplicated in `src/plugins/foxPage/hooks/useOtherOpportunities.ts`, which might signal a need for an abstraction. Fine for now.

Reply:
Definitely! As a follow-up, there are a few things here that I want to abstracted into their own hook, but given something something hooks not singleton, I would prefer to get this one in stable for now and tackle the abstraction in isolation